### PR TITLE
Parse JSDoc comments for ES6 class constructors and methods

### LINKED ISSUE #6646: JSDoc does not appear to work for ES6 class ctors
_From @alexandrudima on January 27, 2016 14:54_

Testing #2218 

`a` is not recognised as a number.

```javascript
"use strict";

class Something {
	
	/**
	 * @param {number} a
	 */
	constructor(a, b) {
		a.
	}
}
```

_Copied from original issue: Microsoft/vscode#2460_

### PR DESCRIPTION
**Pull request by RyanCavanaugh**
Fixes #6646

### CODE REVIEW

File: tests/cases/fourslash/getJavaScriptCompletions16.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+////     /**
+////      * @param {number} a
+////      */
+////     constructor(a, b) {
+////         a./**/
+////     }
+//// }
+
+goTo.marker();
+verify.completionListContains('toFixed', undefined, undefined, 'method');
```
Comment (billti) [inline tests/cases/fourslash/getJavaScriptCompletions16.ts]: Could we also verify that at a `new` site you get the correct intellisense/signature help? (i.e. `var x = new Something(/* here */)`)

Comment (billti) [inline tests/cases/fourslash/getJavaScriptCompletions16.ts]: Should also add a simple test for method params...

**Comment by mhegazy** [comment]
:+1:

File: tests/cases/fourslash/getJavaScriptCompletions16.ts (hunk 1 above)
Comment (RyanCavanaugh) [inline tests/cases/fourslash/getJavaScriptCompletions16.ts]: Done

**Comment by billti** [comment]
Thanks! :+1:

**Pull request merged**